### ISSUE TITLE
Structs vs. single parameters in C API

### ISSUE THREAD

**Issue by t-sommer** [modelica/fmi-standard#667]
Some functions of the C API uses single parameters e.g.

```C
fmi3Status fmi3SetupExperiment(fmi3Instance instance,
                                           fmi3Boolean toleranceDefined,
                                           fmi3Float64 tolerance,
                                           fmi3Float64 startTime,
                                           fmi3Boolean stopTimeDefined,
                                           fmi3Float64 stopTime);
```

others use structs e.g.                 
                                           
```C
fmi3Status fmi3NewDiscreteStates(fmi3Instance instance,
                                             fmi3EventInfo* eventInfo);
```

and some use a combination of both e.g.

```C
fmi3Instance fmi3Instantiate(fmi3String        instanceName,
                                 fmi3InterfaceType fmuType,
                                 fmi3String        fmuInstantiationToken,
                                 fmi3String        fmuResourceLocation,
                                 const fmi3CallbackFunctions* functions,
                                 fmi3Boolean       visible,
                                 fmi3Boolean       loggingOn,
                                 const fmi3CoSimulationConfiguration* fmuCoSimulationConfiguration);
```

Both approaches have their advantages:

Structs

- can be extended in later versions
- reduce the number of parameters

single arguments

- make clear what "goes in" and what "comes out" of the function
- are consistent (no mix of structs and single parameters)
- are easier to implement (esp. in languages that have no C interface, e.g. Java or Python)

IMHO we should consider whether we really need structures as parameters and if so, come up with a rule when to use (and how to extend) them.

**Comment by chrbertsch** [comment]
Regular Design meeting: 

Pierre: I do not have a strong opinion. If I would be redoing it, I would try to avoid structs. Structs have the benefit, that one could add elements in the future.
Torsten S: for EventInfo, we have In- and Out-arguments which is not nice

--> Discuss at F2F meeting

**Comment by andreas-junghanns** [comment]
The initial reason for structs was that we could simply extend without changing the function signature. This turned out to be a hollow reason - we never really used this to prevent API changes.

I would suggest to be consistent and remove all structures.

**Comment by chrbertsch** [comment]
FMI Design Meeting: 

Andreas J: one reason to have the structure was the idea to have backwards compatibility by the possibility to to extend the structure. We do not consider this argument valid any more
We could inline them all
Klaus: logical grouping could be a reason for grouping
Torsten S: We have struct of callbacks that contains also non-callbacks
Andreas: Shall we restructure it?
Klaus: In the C-Style one would not have structures
Andreas: With structures the compilers have more freedom what to do. It usually works. But it could create new problems.
Torsten B: I thought we reuse the structures. But we do not.
Torsten S.: There is no single re-use
Torsten S: We have some arguments in instantiation that do not apply to all FMU types..
Currently we have an enumeration to tell the type.
This is only indirectly related

Poll 1: do we remove structs from the API and inline them into the function call
Yes:5
No:0
Abstain: 7
--> accepted


Poll 2: do we remove all structs from the API and inline them into the function call
Yes:7
No:0
Abstain: 3
--> accepted

--> @t-sommer : I will to it

**Comment by chrbertsch** [comment]
fixed with merged PR #821
